### PR TITLE
Core/NetPlayClient: Map Wiimotes like GCPads

### PR DESCRIPTION
**Pull request by spycrab**

### CODE REVIEW

File: Source/Core/Core/NetPlayClient.cpp [hunk 1]
```diff
@@ -1232,6 +1234,31 @@ int NetPlayClient::LocalPadToInGamePad(int local_pad) const
   return ingame_pad;
 }
 
+int NetPlayClient::NumLocalWiimotes() const
```
Comment (Warepire) [inline Source/Core/Core/NetPlayClient.cpp]: This function is unused.

File: Source/Core/Core/NetPlayClient.cpp [hunk 2]
```diff
@@ -1232,6 +1234,31 @@ int NetPlayClient::LocalPadToInGamePad(int local_pad) const
...
+}
+
+int NetPlayClient::InWiimoteToLocalWiimote(int ingame_wiimote) const
+{
+  // not our pad
+  if (m_wiimote_map[ingame_wiimote] != m_local_player->pid)
+    return 4;
+
+  int local_wiimote = 0;
+  int wiimote = 0;
```
Comment (Warepire) [inline Source/Core/Core/NetPlayClient.cpp]: Why is this one outside the for-loop?

File: Source/Core/Core/NetPlayClient.cpp [hunk 3]
```diff
@@ -1232,6 +1234,31 @@ int NetPlayClient::LocalPadToInGamePad(int local_pad) const
...
+  return static_cast<int>(
+      std::count_if(m_wiimote_map.begin(), m_wiimote_map.end(),
+                    [this](auto mapping) { return mapping == m_local_player->pid; }));
+}
+
+int NetPlayClient::InWiimoteToLocalWiimote(int ingame_wiimote) const
+{
+  // not our pad
+  if (m_wiimote_map[ingame_wiimote] != m_local_player->pid)
+    return 4;
```
Comment (Warepire) [inline Source/Core/Core/NetPlayClient.cpp]: This is rather magical and difficult to comprehend. Also the result is never checked for, anything bad that can happen if more than one ingame_wiimote isn't local?

**Comment by JMC47** [comment]
I attempted to setup Netplay in a more conventional sense and both host and client instantly crashed on start game.

**Comment by mimimi085181** [comment]
Thank you for working on this. The biggest problem i saw with properly implementing this were the attachments. Creating a network packet with the info might be something i could handle, but then you need to use a temporary config for the session attachments setup. Polling the data for a nunchuck is different from a classic controller, so i think the switch should happen while polling or on the config level. And as far as i can remember the reporting mode can change for the same attachments, so in the worst case, 2 different nunchucks in different slots could use different reporting modes.

**Comment by spycrab** [comment]
I have to redo this at some point, it's pretty useless for now though so closing.

**Pull request closed without merging**